src/libpriv/rpmostree-core.c

```diff
@@ -330,13 +330,19 @@ rpmostree_treespec_new_from_keyfile (GKeyFile   *keyfile,
   }
 
   if (!add_canonicalized_string_array (&builder, "packages", NULL, keyfile, error))
-    return FALSE;
+    return NULL;
 
-  if (!add_canonicalized_string_array (&builder, "repos", NULL, keyfile, error))
-    return FALSE;
+  /* We allow the "repo" key to be missing. This means that we rely on hif's
```

Comment:
I'm mildly hestitant about this for the `container` builtin, but we can revisit that later.